Model/AddressResolver.php

```diff
@@ -294,18 +298,30 @@ private function getCityId(AmwalAddressInterface $amwalAddress): ?string
             return null;
         }
 
+        $locale = $this->localeResolver->getLocale();
+
         $connection = $this->resourceConnection->getConnection();
         $tableName = $this->resourceConnection->getTableName('directory_country_region_city');
-
+        $localCityTableName = $this->resourceConnection->getTableName('directory_country_region_city_name');
         if (!$connection->isTableExists($tableName)) {
```

Comment:
As an extra safety I would also add a check for the table specified in `$localCityTableName`:

```suggestion
        if (!$connection->isTableExists($tableName) || !$connection->isTableExists($localCityTableName)) {
```